tests/main_test.py

```diff
@@ -194,46 +198,47 @@ def test_create_group(self):
-        assert content['data']['admin'] == self.user['data']['sub']
+        assert response.json()['data']['admin'] == self.user['data']['sub']
 
         # get the group
-        content, status_code = self.do_post('/group/info', {'id': self.group['_id']['$oid']})
-        assert status_code == 200
-        assert self.group == content['data']
+        response = self.do_post('/group/info', {'id': self.group['_id']['$oid']})
+        self.ensure_status_code_msg(response, 200, "Group successfully retrieved.")
+        assert self.group == response.json()['data']
+
 
         # check persons groups
-        content, status_code = self.do_post('/user/info', {'sub': self.user['data']['sub']})
-        assert status_code == 200
-        user = content['data']
+        response = self.do_post('/user/info', {'sub': self.user['data']['sub']})
+        self.ensure_status_code_msg(response, 200, "User successfully retrieved.")
+        user = response.json()['data']
         assert self.group['_id'] in user['groups']
 
         # update normal group stuff
         data = {
             'name': 'test group update name',
             'desc': 'test group update description'
         }
-        content, status_code = self.do_post('/group/update', {'id': self.group['_id']['$oid'], 'data': data})
-        assert status_code == 200
+        response = self.do_post('/group/update', {'id': self.group['_id']['$oid'], 'data': data})
+        self.ensure_status_code_msg(response, 200, "Group successfully updated.")
 
         # get the group
-        content, status_code = self.do_post('/group/info', {'id': self.group['_id']['$oid']})
-        assert status_code == 200
-        assert content['data']['name'] == 'test group update name'
-        assert content['data']['desc'] == 'test group update description'
+        response = self.do_post('/group/info', {'id': self.group['_id']['$oid']})
+        self.ensure_status_code_msg(response, 200, "Group successfully retrieved.")
+        assert response.json()['data']['name'] == 'test group update name'
+        assert response.json()['data']['desc'] == 'test group update description'
 
         # delete the group
-        content, status_code = self.do_post('/group/delete', {'id': self.group['_id']['$oid']})
-        assert status_code == 200
+        response = self.do_post('/group/delete', {'id': self.group['_id']['$oid']})
+        self.ensure_status_code_msg(response, 200, "Group successfully deleted.")
 
         # check persons groups
-        content, status_code = self.do_post('/user/info', {'sub': self.user['data']['sub']})
-        assert status_code == 200
-        assert self.group['_id'] not in content['data']['groups']
+        response = self.do_post('/user/info', {'sub': self.user['data']['sub']})
+        self.ensure_status_code_msg(response, 200, "User successfully retrieved.")
+        assert self.group['_id'] not in response.json()['data']['groups']
 
     def test_invite(self):
```

Comment:
There's two different tests made here (invite while creating the group, and also invite after creating the group with /group/invite)
Separate them into two different test cases (two different functions)